From 33bd1b170fe23fd0c5f5e797d492fc2a0420128d Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 15 Jan 2010 16:00:42 +0800
Subject: [PATCH 144/449] ENGR00120054 vpu: Remove to set
 ENC_SET_SEARCHRAM_PARAM command on MX51

ME Search ram for encoder will be set in vpu lib, so user cannot set
it by ENC_SET_SEARCHRAM_PARAM command on mx51.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/enc.c | 27 ++++-----------------------
 1 file changed, 4 insertions(+), 23 deletions(-)

diff --git a/test/mxc_vpu_test/enc.c b/test/mxc_vpu_test/enc.c
index 56400b5..b9f8948 100644
--- a/test/mxc_vpu_test/enc.c
+++ b/test/mxc_vpu_test/enc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2010 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -571,35 +571,16 @@ encoder_configure(struct encode *enc)
 	EncInitialInfo initinfo = {0};
 	RetCode ret;
 	MirrorDirection mirror;
-	iram_t iram;
-	int ram_size;
 
 	if (cpu_is_mx27()) {
 		search_pa.searchRamAddr = 0xFFFF4C00;
-	} else if (cpu_is_mx51()) {
-		memset(&iram, 0, sizeof(iram_t));
-		if (enc->cmdl->rot_angle == 90 || enc->cmdl->rot_angle == 270)
-			ram_size = ((enc->picheight + 15) & ~15) * 36 + 2048;
-		else
-			ram_size = ((enc->picwidth + 15) & ~15) * 36 + 2048;
-		IOGetIramBase(&iram);
-		if ((iram.end - iram.start) < ram_size) {
-			err_msg("vpu iram is less than needed.\n");
+		ret = vpu_EncGiveCommand(handle, ENC_SET_SEARCHRAM_PARAM, &search_pa);
+		if (ret != RETCODE_SUCCESS) {
+			err_msg("Encoder SET_SEARCHRAM_PARAM failed\n");
 			return -1;
-		} else {
-			/* Allocate max iram for vpu encoder search ram*/
-			ram_size = iram.end - iram.start;
-			search_pa.searchRamAddr = iram.start;
-			search_pa.SearchRamSize = ram_size;
 		}
 	}
 
-	ret = vpu_EncGiveCommand(handle, ENC_SET_SEARCHRAM_PARAM, &search_pa);
-	if (ret != RETCODE_SUCCESS) {
-		err_msg("Encoder SET_SEARCHRAM_PARAM failed\n");
-		return -1;
-	}
-
 	if (enc->cmdl->rot_en) {
 		vpu_EncGiveCommand(handle, ENABLE_ROTATION, 0);
 		vpu_EncGiveCommand(handle, ENABLE_MIRRORING, 0);
-- 
1.8.0

